tests/integration/test_integration.py

```diff
@@ -937,6 +937,35 @@ async def test_given_vault_b_is_deployed_and_unsealed_when_scale_up_then_status_
             idle_period=5,
         )
 
+    @pytest.mark.abort_on_fail
+    async def test_given_vault_b_is_deployed_and_unsealed_when_scale_down_then_up_then_status_is_active(
```

Comment:
I'm assuming it going to active means that the raft storage remains intact and this test passes like this.

Reply:
I just realised we explicitly delete vault data using `_delete_vault_data` when a unit is removed.